### ISSUE TITLE
[java] DanglingJavadoc false positive on record compact constructors

### ISSUE THREAD

**Issue by cowwoc** [pmd/pmd#6103]
**Affects PMD Version:** 7.17.0

**Rule:**

DanglingJavadoc
<https://docs.pmd-code.org/pmd-doc-7.17.0/pmd_rules_java_documentation.html#danglingjavadoc>

**Description:**

PMD's DanglingJavadoc rule incorrectly reports Javadoc comments on Java record compact constructors as "dangling" (not belonging to any class, method, or field). Compact constructors are valid constructor declarations introduced in Java 14 (JEP 359) and should be recognized as valid Javadoc targets.

**Code Sample demonstrating the issue:**

```java
package test;

import static io.github.cowwoc.requirements12.java.DefaultJavaValidators.requireThat;

/**
 * Represents a position in source code with line and column information.
 *
 * @param line the line number (1-based)
 * @param column the column number (1-based)
 */
public record SourcePosition(int line, int column)
{
    /**
     * Compact constructor validating line and column are positive.
     */
    public SourcePosition
    {
        requireThat(line, "line").isPositive();
        requireThat(column, "column").isPositive();
    }
}
```

**Expected outcome:**

PMD should recognize the Javadoc comment as documenting the compact constructor and not report any violation.

**Actual outcome:**

PMD reports a violation at line 15:
```
Javadoc comment does not belong to any class, method or field
```

This is a false positive. The Javadoc comment documents the compact constructor `public SourcePosition { ... }`, which is a valid Java language construct that deserves documentation.

**Technical Context:**

Java records support compact constructor syntax where the parameter list is omitted:

```java
public record Point(int x, int y) {
    // Compact constructor - no parameter list needed
    public Point {
        if (x < 0 || y < 0) throw new IllegalArgumentException();
        // Field assignments (this.x = x; this.y = y;) are implicit
    }
}
```

This is equivalent to the canonical constructor form:

```java
public record Point(int x, int y) {
    // Canonical constructor - explicit parameters
    public Point(int x, int y) {
        if (x < 0 || y < 0) throw new IllegalArgumentException();
        this.x = x;
        this.y = y;
    }
}
```

PMD's DanglingJavadoc rule should recognize `ASTCompactConstructorDeclaration` as a valid Javadoc target, similar to how it recognizes regular constructor and method declarations.

**Running PMD through:** Maven (maven-pmd-plugin 3.25.0)

**Comment by zbynek** [comment]
Thanks, I think I know where the problem is, I hope to have a PR ready soon.
